File: LeetCode/cpp/988.cpp
class Solution {
public:
    vector<string> vec;

    string smallestFromLeaf(TreeNode *root) {
        string ret;
        if (!root)
            return ret;
        FindString(root, ret);
        for (auto &s:vec)
            std::reverse(s.begin(), s.end());
        ret = vec[0];
        for (int i = 1; i < vec.size(); ++i)
            ret = min(ret, vec[i]);
        return ret;
    }

    void FindString(TreeNode *node, string &curr) {
        curr += ('a' + node->val);
        if (!node->left && !node->right)
            vec.push_back(curr);
        if (node->left)
            FindString(node->left, curr);
        if (node->right)
            FindString(node->right, curr);
        curr.pop_back();
    }


};

File: LeetCode/python3/988.py
class Solution:
    def smallestFromLeaf(self, root: 'TreeNode') -> 'str':
        if not root:
            return ""
        self.ret = "z" * 1000
        self.Traverse(root, "")
        return self.ret


    def Traverse(self, node, st):
        st += chr(97 + node.val)
        if(not node.left and not node.right):
            if st[::-1] < self.ret:
                self.ret = st[::-1]
        if node.left:
            self.Traverse(node.left, st)
        if node.right:
            self.Traverse(node.right, st)
        st = st[:-1]
